From 24274aa7ab55b65ef0228058a0ff7d53f40e42f8 Mon Sep 17 00:00:00 2001
From: Jayesh Choudhary <j-choudhary@ti.com>
Date: Fri, 7 Feb 2025 11:09:03 +0530
Subject: [PATCH 17/68] PENDING: arm64: dts: ti: k3-j721s2-main: Convert
 edp_serdes_mux node into reg-mux

Convert mmio-mux to reg-mux to remove the dependency of the
parent node being syscon node. Also, change mask as required.

Now changing the scm_bus to "simple-bus" compatible won't
cause mux-controller failure.

Signed-off-by: Jayesh Choudhary <j-choudhary@ti.com>
---
 arch/arm64/boot/dts/ti/k3-j721s2-main.dtsi | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/ti/k3-j721s2-main.dtsi b/arch/arm64/boot/dts/ti/k3-j721s2-main.dtsi
index 2612dd9c8684..d29a8fb3554e 100644
--- a/arch/arm64/boot/dts/ti/k3-j721s2-main.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-j721s2-main.dtsi
@@ -78,11 +78,11 @@ ehrpwm_tbclk: clock-controller@140 {
 		};
 
 		edp_serdes_mux: mux-controller@310 {
-			compatible = "mmio-mux";
+			compatible = "reg-mux";
 			reg = <0x310 0x4>;
 			#mux-control-cells = <1>;
 			/* EDP0 to SERDES0 lane 0/1 or 2/3 mux */
-			mux-reg-masks = <0x310 0x10000000>;
+			mux-reg-masks = <0x0 0x10000000>;
 		};
 	};
 
-- 
2.47.2

